internal/service/ch/ch.go

```diff
@@ -149,3 +149,24 @@ func (s *Service) getAggSignals(ctx context.Context, stmt string, args []any) ([
 	}
 	return signals, nil
 }
+
+// GetAvailableSignals returns a slice of available signals from the ClickHouse database.
+// if no signals are available, a nil slice is returned.
```

Comment:
`nil` is an interesting choice. It does match the behavior of something like [`FindAll`](https://pkg.go.dev/regexp#Regexp.FindAll) but at the GraphQL level (and it does seem like it gets propagated all the way up) it seems a little strange to me.

Reply:
Is it part of the contract that they're alphabetical by name?

Reply:
> nil is an interesting choice. It does match the behavior of something like [FindAll](https://pkg.go.dev/regexp#Regexp.FindAll) but at the GraphQL level (and it does seem like it gets propagated all the way up) it seems a little strange to me.

For GQL, I thought this would be the desired behavior since we are leveraging the NULL property. A User would simply check if the value exists instead of having to check if the list is of length 0.

I'm not sure what you mean by contract, but I `ORDER BY name` so we have consistent results.

Reply:
> I'm not sure what you mean by contract

Like “is it worth mentioning in the doc string”.

Reply:
`signals` appears to act the same way 
vis-à-vis `[]` versus `null`. It’s a bit surprising to me but I’m good with it, let’s keep it.